packages/terra-button/src/Button.jsx

```diff
@@ -308,9 +308,9 @@ class Button extends React.Component {
     );
 
     // ignore aria-label for button with decorative icons.
-    let ariaLabel = (iconType !== IconTypes.DECORATIVE) ? customProps['aria-label'] : undefined;
-    if ((isIconOnly || variant === 'utility') && iconType === IconTypes.INFORMATIVE) {
-      ariaLabel = ariaLabel || text;
+    let ariaLabel = customProps['aria-label'] ? customProps['aria-label'] : text;
```

Comment:
this will add aria-label to all variants. where as existing implementation was to add aria-label only when it is provided for other variants except iconOnly and utility. for iconOnly and utility it will add the area label even when consumer does not pass it.